src/controller/TypedReadCallback.h

```diff
@@ -82,8 +82,7 @@ class TypedReadAttributeCallback final : public app::ReadClient::Callback
         VerifyOrExit(aPath.mClusterId == mClusterId && aPath.mAttributeId == mAttributeId, err = CHIP_ERROR_SCHEMA_MISMATCH);
         VerifyOrExit(apData != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
 
-        err = app::DataModel::Decode(*apData, value);
-        SuccessOrExit(err);
+        SuccessOrExit(app::DataModel::Decode(*apData, value));
```

Comment:
This is wrong, because it's no longer setting `err`, so we will not call `mOnError`.... I am surprised we have no tests for this.  :(